lib/cfn-nag/custom_rules/AmazonMQBrokerUserPasswordRule.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require 'cfn-nag/util/enforce_reference_parameter'
+require 'cfn-nag/util/enforce_string_or_dynamic_reference'
+require_relative 'base'
+
+class AmazonMQBrokerUserPasswordRule < BaseRule
+
+  def rule_text
+    'Amazon MQ Broker User Password property should not show password in plain text, resolve an unsecure ssm string, or have a default value for parameter.'
+  end
+
+  def rule_type
+    Violation::FAILING_VIOLATION
+  end
+
+  def rule_id
+    'F52'
+  end
+
+  def resource_type
+    'AWS::AmazonMQ::Broker'
+  end
+
+  def audit_impl(cfn_model)
+    resources = cfn_model.resources_by_type('AWS::AmazonMQ::Broker')
+
+    violating_resources = resources.select do |resource|
+      violating_users = resource.users.select do |user|
+        insecure_parameter?(cfn_model, user['Password']) || \
```

Comment:
1. break this logic out into a predicate.  the backslashes give me the willies, but aside from that a complex boolean like this deserves its own named method to lower CCM and improve readability

2. checking the nil.... i'm hemming and hawing over that.  cfn-model is the underlying gem that is supposed to parse and prepare things so that we don't have to do a lot of this checking in nag rules.  since this is a nested data structure though (User) we are stuck doing some of that work here.  i guess i'd suggest checking it's there before trying to reference it... but then if it's not there, just do nothing.